src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ProvideStreamMemoryBasedAsyncOverrides.cs

```diff
@@ -0,0 +1,208 @@
+                    violatingType.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                    arrayBasedOverride.OverriddenMethod.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                    memoryBasedMethod.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat));
+            }
+        }
+
+        private static bool TryGetRequiredSymbols(Compilation compilation, out RequiredSymbols requiredSymbols)
+        {
+            var int32Type = compilation.GetSpecialType(SpecialType.System_Int32);
+            var byteType = compilation.GetSpecialType(SpecialType.System_Byte);
+            var byteArrayType = compilation.CreateArrayTypeSymbol(byteType);
+            var memoryOfByteType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemMemory1)?.Construct(byteType);
+            var readOnlyMemoryOfByteType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemReadOnlyMemory1)?.Construct(byteType);
+            var streamType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemIOStream);
+            var cancellationTokenType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingCancellationToken);
+
+            if (memoryOfByteType is null || readOnlyMemoryOfByteType is null || streamType is null || cancellationTokenType is null)
+            {
+                requiredSymbols = default;
+                return false;
+            }
+
+            var readAsyncArrayMethod = GetOverload(streamType, ReadAsyncName, byteArrayType, int32Type, int32Type, cancellationTokenType);
+            var readAsyncMemoryMethod = GetOverload(streamType, ReadAsyncName, memoryOfByteType, cancellationTokenType);
+            var writeAsyncArrayMethod = GetOverload(streamType, WriteAsyncName, byteArrayType, int32Type, int32Type, cancellationTokenType);
+            var writeAsyncMemoryMethod = GetOverload(streamType, WriteAsyncName, readOnlyMemoryOfByteType, cancellationTokenType);
+
+            if (readAsyncArrayMethod is null || readAsyncMemoryMethod is null || writeAsyncArrayMethod is null || writeAsyncMemoryMethod is null)
+            {
+                requiredSymbols = default;
+                return false;
+            }
+
+            requiredSymbols = new RequiredSymbols(
+                streamType, memoryOfByteType, readOnlyMemoryOfByteType,
+                readAsyncArrayMethod, readAsyncMemoryMethod,
+                writeAsyncArrayMethod, writeAsyncMemoryMethod);
+            return true;
+        }
+
+        private static IMethodSymbol? GetOverload(ITypeSymbol containingType, string methodName, params ITypeSymbol[] argumentTypes)
+        {
+            return containingType.GetMembers(methodName)
+                .SingleOrDefault(symbol => symbol is IMethodSymbol m && IsMatch(m, argumentTypes)) as IMethodSymbol;
+
+            static bool IsMatch(IMethodSymbol method, ITypeSymbol[] argumentTypes)
```

Comment:
@Evangelink Was there a helper method that does this?

Reply:
The helper method requires instances of `ParameterInfo`, which themselves require `INamedTypeSymbol` instances. It was significantly shorter to write it out my way. I did originally write it out using the helper method, but it was significantly longer and more verbose.